src/cmd/services/m3coordinator/ingest/carbon/ingest.go

```diff
@@ -106,7 +112,23 @@ func NewIngester(
 		return nil, err
 	}
 
+	poolOpts := pool.NewObjectPoolOptions().
+		SetInstrumentOptions(opts.InstrumentOptions).
+		SetRefillLowWatermark(0).
+		SetRefillHighWatermark(0).
+		SetSize(defaultResourcePoolSize)
+
+	resourcePool := pool.NewObjectPool(poolOpts)
+	resourcePool.Init(func() interface{} {
+		return lineResources{
```

Comment:
Think it's better to make this `&lineResources{...}` just in case when it gets boxed back to `interface{}` during the `putLineResources(...)` method it gets reallocated on the heap (even though it would just be the wrapping struct itself needing to be reallocated)?

Reply:
benchmarked it and you're right, will fix